addons/website_form/static/src/snippets/s_website_form/options.js

```diff
@@ -821,6 +830,21 @@ options.registry.WebsiteFieldEditor = FieldEditor.extend({
         // We need to reload the existing type list.
         this.rerender = true;
     },
+    /**
+     * Generates new ids to avoid having several times the same id
+     * 
+     * @override
+     */
+    onClone() {
+        const originalInputEl = this.$target[0].querySelector('[id]:not(div)');
+        const cloneContainerEl = document.querySelectorAll(`#${CSS.escape(originalInputEl.id)}`)[1].closest('.s_website_form_field');
+        for (const el of cloneContainerEl.querySelectorAll('[id]:not(div)')) {
+            const newId = this._generateUniqueID();
+            cloneContainerEl.querySelector(`label[for="${el.id}"]`).setAttribute('for', newId);
+            el.id = newId;
+        }
+        this.$target.click(); // We are sure that cleanForSave will be executed on the new element.
```

Comment:
See GDI's comment on the PR but this is definitely not the right solution. Either cleanForSave is not called while it should and it's a bug to solve at the snippet editor level, either what is done in that cleanForSave should not be done there.